pkg/sqlmodel/reduce.go

```diff
@@ -58,6 +55,41 @@ func (r *RowChange) IdentityValues() ([]interface{}, []interface{}) {
 	return pre, post
 }
 
+// RowIdentity returns the identity of this row change, caller should
+// call IsIdentityUpdated/SplitUpdate before calling this method to
+// make sure it's not updating the identity itself.
+// we extract identity from preValues for update/delete, postValues for insert.
+// if there's no primary key, return all values.
```

Comment:
Does it mean we can identify rows without PK (so that we can remove the `WithoutPKError`)?

Reply:
whether row has PK is determined by the tableinfo, it will be skipped if there no PK

where's WithoutPKError?

Reply:
I mean we can validate the rows without PK (instead of skipping them) because this function helps identifying rows thought they don't have PK.

Reply:
this method may used in other place, `if there's no primary key, return all values.` is just a workaround when there's no real identity.

cannot validate pk-less table, see 2 same rows in upstream, only 1 in downstream, validation will success. and validate pk-less table put a lot of load on downstream.